src/Mod/Part/App/PartFeature.h

```diff
@@ -171,6 +171,17 @@ class PartExport Feature : public App::GeoFeature
 
     void registerElementCache(const std::string &prefix, PropertyPartShape *prop);
 
+    /** Helper function to obtain mapped and indexed element name from a shape
+     * @params shape: source shape
+     * @param name: the input name, can be either mapped or indexed name
+     * @return Returns both the indexed and mapped name
+     *
+     * If the 'name' referencing a non-primary shape type, i.e. not
+     * Vertex/Edge/Face, this function will auto generate a name from primary
+     * sub-shapes.
+     */
+    std::pair<std::string,std::string> getExportElementName(TopoShape shape, const char *name) const;

```

Comment:
Returning a pair of two identical types is pretty opaque. Can we go with a struct with named elements instead?

Reply:
Careful what you ask for:  #15535